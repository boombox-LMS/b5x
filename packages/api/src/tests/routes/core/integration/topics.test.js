const supertest = require("supertest");

describe("Topic registration flow is operational", () => {
  let app;
  const apiPrefix = global.__SUPERTEST_API_PREFIX__;
  let initialTopicCount;

  // Add a seeded test database to the app, available as app.get('db')
  // TODO: Make it possible to seed with no users at all, since we don't actually need any here
  beforeAll(async () => {
    app = await global.buildAppWithSeededTestDatabase({
      users: [{ persona: "completed" }],
    });
    initialTopicCount = await app.get("db").topics.count();
  });

  // Close the database connection, which allows Jest to exit gracefully
  afterAll(async () => {
    await app.get("db").destroy();
  });

  test("topics.register allows registration of unique slug", async () => {
    await supertest(app)
      .post(apiPrefix + `topics.register`)
      .send({ slug: "brand-new-topic-slug" })
      .expect(200)
      .then((res) => {
        expect(res.body).toMatchSnapshot();
      })
      .catch((e) => {
        throw e;
      });
  });

  test("topics.register creates a new topic when given a unique slug", async () => {
    const newTopicCount = await app.get("db").topics.count();
    expect(newTopicCount).toEqual(initialTopicCount + 1);
  });

  test("topics.register rejects duplicate slug registration", async () => {
    await supertest(app)
      .post(apiPrefix + `topics.register`)
      .send({ slug: "brand-new-topic-slug" })
      .expect(200)
      .then((res) => {
        expect(res.body).toMatchSnapshot();
      })
      .catch((e) => {
        throw e;
      });
  });
});
