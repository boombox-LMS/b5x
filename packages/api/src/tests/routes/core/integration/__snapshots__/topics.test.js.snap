// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topic registration flow is operational topics.register allows registration of unique slug 1`] = `
{
  "succeeded": true,
}
`;

exports[`Topic registration flow is operational topics.register rejects duplicate slug registration 1`] = `
{
  "error": "Topic slug brand-new-topic-slug is already in use",
  "succeeded": false,
}
`;
